src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -2580,6 +2582,14 @@ PruneShardList(Oid relationId, Index tableId, List *whereClauseList,
 	Var *partitionColumn = PartitionColumn(relationId, tableId);
 	char partitionMethod = PartitionMethod(relationId);
 
+	/* short circuit for reference tables */
+	if (partitionMethod == DISTRIBUTE_BY_NONE)
+	{
+		Assert(list_length(shardIntervalList) == 1);
```

Comment:
I feel like we tend to overuse Assert. Incorrect metadata shouldn't really crash the server, but rather cause errors (e.g. someone may have a bug in a metadata-changing script). We should error out somewhere if a reference table has >1 shard, though probably closer to where we load shard list.